schema/manager.go

```diff
@@ -136,6 +136,68 @@ func (manager *Manager) OrderedSchemas() []*Schema {
 	return res
 }
 
+//reorderSchema tries reorder schemas using Tarjan's algorithm
+type state int
+
+const (
+	notVisited state = iota
+	visited
+	temporaryVisited
+)
+
+func visitSchema(schemas []*Schema, schemaList []string, schema *Schema, visitedSchema map[string]state) ([]string, error) {
+	if visitedSchema[schema.ID] == temporaryVisited {
+		return nil, fmt.Errorf("Schemas aren't DAG. We can't reorder automatically")
+	}
+	if visitedSchema[schema.ID] == visited {
+		return schemaList, nil
```

Comment:
This is from Tarjan algorithm
https://en.wikipedia.org/wiki/Tarjan%27s_strongly_connected_components_algorithm